dask_cuda/utils.py

```diff
@@ -972,3 +988,41 @@ def convert(self, value, param, ctx):
                 choices_str = ", ".join(f"'{c}'" for c in self.choices)
                 self.fail(f"invalid choice(s): {v}. (choices are: {choices_str})")
         return values
+
+
+def _get_active_ucx_implementation_name(protocol):
+    """Get the name of active UCX implementation.
+
+    Determine what UCX implementation is being activated based on a series of
+    conditions. UCXX is selected if:
+    - The protocol is `"ucxx"`, or the protocol is `"ucx"` and the `distributed-ucxx`
+      package is installed.
+    UCX-Py is selected if:
+    - The protocol is `"ucx-old"`, or the protocol is `"ucx"` and the `distributed-ucxx`
+      package is not installed, in which case a `FutureWarning` is also raised.
+
+    Parameters
+    ----------
+    protocol: str
+        The communication protocol selected.
+
+    Returns
+    -------
+    The selected implementation type, either "ucxx" or "ucx-old".
+
+    Raises
+    ------
+    ValueError
+        If protocol is not a valid UCX protocol.
+    """
+    has_ucxx = bool(importlib.util.find_spec("distributed_ucxx"))
```

Comment:
I think this might be a more canonical way to write this check:
```suggestion
    has_ucxx = importlib.util.find_spec("distributed_ucxx") is not None
```